tests/metrics/collect_mem_consmd.sh

```diff
@@ -0,0 +1,299 @@
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#  Description of the test:
+#  The objective of this tool is to measure the average memory consumption
+#  of a Clear Containers component during a determined Clear Containers
+#  stress flow.
+
+# set -x
+# set -e
+
+# General env
+MYSELF=${0##*/}
+CC_NUM=5
+MSGS_NUM=10000
+MSG_SIZE=80
+SHARED_DIR="shared"
+PROC="cc-proxy"
+
+# Host/Container paths
+HOST_SHARED_PATH="$(pwd)/$SHARED_DIR"
+CONT_SHARED_PATH="/root/$SHARED_DIR"
+
+# Aux files
+HOST_GATE="$HOST_SHARED_PATH/gate"
+CONT_GATE="$CONT_SHARED_PATH/gate"
+HOST_RDM_FILE="$HOST_SHARED_PATH/out.data"
+CONT_RDM_FILE="$CONT_SHARED_PATH/out.data"
+CSV_FILE="mem.csv"
+
+declare WORKLOAD
+declare COLLECT_PID
+declare F_OPTS
+declare AVG
+
+# Shows help about this script
+function help()
+{
+usage=$(cat << EOF
+Usage: $MYSELF [-h] [--help] [-v] [--version]
+  Description:
+       This tool takes memory consumption
+       information about a determined number
+       of containers.
+  Options:
+       -h, --help             Help page
+       -c, --num-containers   Set the number of containers
+                              that will be launched.
+       -m, --num-messages     The number of messages is the
+                              workload of each container.
+       -s, --size-message     The number of characters per
+                              message.
+       -p, --process-filter   This option restricts which
+                              processes are selected for
+			      collection/display tool.
+       -x --exact-match       Get metrics just about exact
+                              match.
+       -v, --version          Show version.
+EOF
+)
+	echo "$usage";
+}
+
+# This workload will run inside of each Clear
+# Container, it will monitor a shared path with
+# host, this is in order to establish communication
+# with host and activate the I/O read/write
+# workload at the same time in all containers.
+function implement_workload()
+{
+	WORKLOAD=$(cat << EOF
+	#!/bin/bash
+
+	function print_rdmsg()
+	{
+		echo "$(cat /dev/urandom 2>/dev/null | \
+			LC_CTYPE=C tr -dc "[:alnum:]" 2>/dev/null | \
+			head -c $MSG_SIZE )"
+	}
+
+	while [ 1 ];do
+		if [ -f "$CONT_GATE" ];then
+			for (( i=0; i<$MSGS_NUM; i++ ));do
+				print_rdmsg
+			done
+			break;
+		else
+			sleep 0.2;
+		fi
+	done
+	exit 0;
+EOF
+)
+}
+
+# This function launch collectl tool
+# in order to get metrics about a specific tool.
+function start_collect()
+{
+	collectl -oTm -sZm -i1:1 \
+		--procfilt f$PROC \
```

Comment:
We have one cc-shim process per container. When PROC=cc-shim, are all the cc-shim processes monitored? if so that means the results is actually the total memory consumed by all the cc-shim processes, not just one. Maybe we should tell that to the user and/or divide the result by the number of shims?